test/e2e/azure_machinepool_drain.go

```diff
@@ -157,13 +166,39 @@ func testMachinePoolCordonAndDrain(ctx context.Context, mgmtClusterProxy, worklo
 			return err
 		}
 
-		decreasedReplicas := *owningMachinePool.Spec.Replicas - int32(1)
+		decreasedReplicas = *owningMachinePool.Spec.Replicas - int32(1)
 		owningMachinePool.Spec.Replicas = &decreasedReplicas
 		return helper.Patch(ctx, owningMachinePool)
 	}, 3*time.Minute, 3*time.Second).Should(Succeed())
 
-	// TODO setup a watcher to validate expected 2nd order drain outcomes
-	// https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2159
+	By(fmt.Sprintf("checking for a machine to start draining for machine pool: %s/%s", amp.Namespace, amp.Name))
+	Eventually(func() error {
+		ampmls, err := getAzureMachinePoolMachines(ctx, mgmtClusterProxy, workloadClusterProxy, amp)
+		if err != nil {
+			LogWarning(errors.Wrap(err, "failed to list the azure machine pool machines").Error())
+			return err
+		}
+
+		for i := range ampmls {
+			if conditions.IsTrue(&ampmls[i], clusterv1.DrainingSucceededCondition) {
+				controllerutil.RemoveFinalizer(&ampmls[i], azureMachinePoolMachineTestFinalizer)
+				return nil // started draining the node prior to delete
+			}
+		}
+
+		return errors.New("no machine has started to drain")
+	}, 3*time.Minute, 3*time.Second).Should(Succeed())
+
+	By(fmt.Sprintf("waiting for the machine pool to scale down to %d replicas: %s/%s", decreasedReplicas, amp.Namespace, amp.Name))
+	Eventually(func() (int32, error) {
+		mp, err := getOwnerMachinePool(ctx, mgmtClusterProxy.GetClient(), amp.ObjectMeta)
+		if err != nil {
+			LogWarning(err.Error())
+			return 0, err
+		}
+
+		return int32(len(mp.Spec.ProviderIDList)), nil
+	}, 3*time.Minute, 3*time.Second).Should(Equal(decreasedReplicas))
```

Comment:
Prob 10 mins is good here as well

Reply:
This one I think we want to bump no matter what, as this can be Azure operations-dependent (deleting a VM)